tests/integration/utils.py

```diff
@@ -0,0 +1,242 @@
+
+
+# Setup base URL from configuration file
+def setup_base_url(config):
+    api = config['api']
+    return api['local_base_url'] if config['local_test'] else api['base_url']
+
+
+# Setup request session from configuration file
+def setup_session(config):
+    session = requests.Session()
+
+    if config['local_test']:
+        basic_auth = config['auth']['basic_auth']
+        session.auth = (basic_auth['username'], basic_auth['password'])
+        session.verify = False
+    else:
+        oauth2 = config['auth']['oauth2']
+        data = {
+            'client_id': oauth2['client_id'],
+            'client_secret': oauth2['client_secret'],
+            'grant_type': 'client_credentials'
+        }
+        try:
+            res = requests.post(url=oauth2['token_api_url'], data=data)
+            token = res.json()['access_token']
+        except KeyError:
+            sys.exit('Error: invalid OAuth2 credentials')
+        session.headers = {'Authorization': f'Bearer {token}'}
+
+    return session
+
+
+# Get response content in JSON format
+def get_json_content(self, response):
+    try:
+        return response.json()
+    except json.decoder.JSONDecodeError:
+        self.fail('Response not in JSON format')
+
+
+# Get resource schema from OpenAPI specification
+def get_resource_schema(self, resource):
+    return self.openapi['definitions'][resource]['properties']
+
+
+# Helper function to make a web request and lightly validate the response
+def make_request(self, endpoint, expected_status_code,
+                 params=None,
+                 max_elapsed_seconds=5):
+    """
+    Keyword arguments:
+    * endpoint -- the endpoint to request
+    * expected_status_code -- expected HTTP status code
+    * params -- [optional] key-value pairs parameters (default: None)
+    * max_elapsed_seconds -- [optional] maximum elapsed times (default: 5)
+    Return:
+    A response object contains a server’s response to an HTTP request
+    """
+
+    requested_url = f'{self.base_url}{endpoint}'
+    response = self.session.get(requested_url, params=params)
+    logging.debug(f'Sent request to {requested_url}')
+    # Response status code should be as expected
+    status_code = response.status_code
+    self.assertEqual(status_code, expected_status_code)
+
+    # Response time should less then max_elapsed_seconds
+    elapsed_seconds = response.elapsed.total_seconds()
+    logging.debug(f'Request took {elapsed_seconds} second(s)')
+    self.assertLess(elapsed_seconds, max_elapsed_seconds)
+
+    return response
+
+
+# Check the schema of response match OpenAPI specification
+def check_schema(self, response, schema):
+    # Mapping of OpenAPI data types and python data types
+    types_dict = {
+        'string': str,
+        'integer': int,
+        'int32': int,
+        'int64': int,
+        'float': float,
+        'double': float,
+        'boolean': bool,
+        'array': list,
+        'object': dict
+    }
```

Comment:
Not your fault, but it looks like `types_dict` handles all of the [Swagger data types](https://swagger.io/docs/specification/data-models/data-types/) except for `number`. Usually, using `type: number` is followed by a `format` like `boolean` but we should handle the case of not having a `format` for this type. I talked with @tsoliangwu0130 and it seems like the best way to handle this is by using the `float` type. Could you add this to `types_dict`? This will also be able to go in the skeleton.